src/net.h

```diff
@@ -176,6 +178,7 @@ extern std::map<CInv, CDataStream> mapRelay;
 extern std::deque<std::pair<int64_t, CInv> > vRelayExpiration;
 extern CCriticalSection cs_mapRelay;
 extern limitedmap<CInv, int64_t> mapAlreadyAskedFor;
+extern limitedmap<CInv, int64_t> mapAlreadyReceived;
```

Comment:
Just a light opinion, why not renaming `mapAlreadyReceived` with something like `mapAlreadyReceivedTxs`? I'd like to make explicit this map refers to txs only, compared to other data-structure like `mapAlreadyAskedFor`, `setAskFor`, `mapAskFor` which work generically on invs.

Reply:
While the description of this PR focuses on the tx type, I'd rather leave it generic. `mapAlreadyReceived` can be used for generic invs, and in fact it could also be used to filter blocks that did not make it into `mapBlockIndex`, for example in case of failed header validation.